src/content/docs/recipes/dynamic-footer-date.mdx

```diff
@@ -0,0 +1,48 @@
+---
+title: Dynamic footer date
+astroRange: ^4.0.0
+---
+
+Having a copyright notice with the current year such as `© 2024 - All rights reserved` is a common
+need (although it doesn't legally enforce anything).
+
+## The problem (maybe not!)
+
+Looking at this solution:
+
+```astro title="src/components/Footer.astro" {2} "year"
+---
+const year = new Date().getFullYear()
+---
+
+<footer>
+    <p>&copy; {year} - All rights reserved</p>
+</footer>
+```
+
+It works but the year will only be refreshed each time you rebuild the site.
+
+## The solution
+
+With a tiny bit of JavaScript we can solve this issue really easily!
+
+```astro title="src/components/Footer.astro" {"1. We prepare a build time fallback": 2-3} {"2. We add a span with an id and pass the fallback": 7-8} {"3. We create the date client-side so it's up to date": 10-18}
+---
+
+const year = new Date().getFullYear()
```

Comment:
clearly something that I lack the concept of: Why is a fallback for getFullYear() there in the first place? Is it that obvious having a fallback for a method [supported by majority of browsers](https://caniuse.com/?search=getfullyear)?
Other than that PR is LGTM 👍

Reply:
my bad it's no clear enough! If the browser can't run JS then the script won't run. This fallback is to make sure it will show the last known date (based on build time). I should definitely clarify that!

Reply:
right i forgot users can turn off JS 🤦 , thanks for letting me know 😄

Reply:
unresolving as i need to be more explicit